rfcs/field-selection.md

```diff
@@ -0,0 +1,1453 @@
+}
+```
+
+This notation implies that the nested field `id` under `address` is semantically equivalent to the argument `userId`.
+
+However, complexities come when multiple fields need to be referenced, particularly when they are nested:
+
+**Multiple Field Reference - Output field is deeper in the tree**
+```graphql
+type Query {
+    findUserByName(user: UserInput! @is(field: "profile { firstName } lastName")): User @lookup
+}
+
+input UserInput {
+    firstName: String!
+    lastName: String!
+}
+```
+
+The challenge here is to clearly define that `User.profile.firstName` is equivalent to `UserInput.firstName`.
+
+To address this, a notation like input object fields could be utilized:
+```graphql
+type Query {
+    findUserByName(user: UserInput! @is(field: "firstName: { profile { firstName } } lastName")): User @lookup
+}
+```
+However, this solution mixes input fields and output fields, potentially leading to confusion in
+interpreting the mappings.
+
+## `FieldSelection` as a path combined with an input object
+Another solution is to treat the `FieldSelection` as a path and borrow a few concepts from the input object syntax to shape the input object.
+
+A single field reference would look exactly the same as in the previous example. 
+
+**Single Field Reference**
+```graphql
+extend type Query {
+    userById(id: ID! @is(field: "id")): User @lookup
+}
+```
+
+Multiple field references follow the same format:
+**Multiple Field Reference**
+```graphql
+extend type Query {
+    findUserByName(user: UserInput! @is(field: "firstName lastName")): User @lookup
+}
+```
+
+To reference a deeper nested field, a path syntax is used:
+**Single Field Nested Reference**
+```graphql
+extend type Query {
+    userById(userId: ID! @is(field: "address.id")): User @lookup
+}
+```
+
+This approach also allows for pointing to fields across different hierarchies:
+
+**Multiple Field Reference - Output field is deeper in the tree**
+```graphql
+type Query {
+    findUserByName(user: UserInput! @is(field: "profile.firstName lastName")): User @lookup
+}
+
+input UserInput {
+    firstName: String!
+    lastName: String!
+}
+```
+
+To handle renaming, simply specify the new name followed by the path to the field:
+```graphql
+type Query {
+    findUserByName(user: UserInput! @is(field: "firstName: profile.name lastName")): User @lookup
+}
+```
+
+The main challenge with this approach is abstract types. A generic path syntax might solve this for some cases:
+
+**Output field is deeper in the tree with abstract type and fieldname is different**
+```graphql
+extend type Query {
+   findUserByName(user: UserInput! @is(field: "firstName: profile<EntraProfile>.name lastName")): User @lookup
+}
+```
+
+However, it fails when multiple abstract types are possible for the same field unless we introduce a
+new syntax to handle this:
+
+```graphql
+extend type Query {
+   mediaById(id: ID! @is(field: "<Movie>.id | <Book>.id")): MovieOrBook @lookup
```

Comment:
With operation expressions this would be:

```suggestion
   mediaById(id: ID! @is(field: "Movie.id Book.id")): MovieOrBook @lookup
```

That would effectively compile out to: `... Movie { id } ... Book { id }`.